Downscale site icon images to 32px in reader view

pfefferle opened the pull request:
This PR refactors the site icon rendering approach by replacing CSS transform scaling with direct image dimension sizing. The change improves icon rendering quality by setting images to 32px directly rather than scaling larger images down, while maintaining the 64px container sizes for proper layout spacing.

| Before | After |
| --- | --- |
| <img width="1255" height="1045" alt="Screenshot 2025-11-24 at 09 51 35" src="https://github.com/user-attachments/assets/bcfdfacb-24fc-44b5-987b-bd4a1cee9fc3" /> | <img width="1272" height="857" alt="Screenshot 2025-11-24 at 10 05 14" src="https://github.com/user-attachments/assets/c7727b83-bb6a-4212-a596-5da717c247da" /> |

## Proposed changes:
* Remove CSS transform scale from site hub icon button
* Add direct 32px sizing to `.site-icon__image` elements in both site hub and actor switcher
* Keep container sizes at 64px for proper layout spacing

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:

* Navigate to the reader view in the Social Web interface
* Verify the site icon in the header displays correctly at 32px without CSS transform
* Switch to site actor mode and verify the avatar in the actor switcher displays correctly at 32px
* Confirm the icon is crisp, properly centered, and maintains proper spacing with surrounding elements
* Test on different screen sizes to ensure responsive behavior

### Changelog entry

No changelog entry required for this PR.

pfefferle committed:
Downscale site icon images to 32px in reader view

Remove CSS transform scale and update icon image dimensions directly to 32px while keeping container sizes at 64px for proper layout spacing.

pfefferle committed:
Merge branch 'add/reader' into feature/downscale-reader-icons

obenland commented:
This should be fixed by adding more specific styles to [the `.site-icon__image` selector in the site-icon component](https://github.com/Automattic/wordpress-activitypub/blob/add/reader/src/social-web/components/site-icon/style.scss#L11-L13):

```css
.site-icon__image {
    aspect-ratio: 1 / 1;
    background: #333;
    height: 100%;
    object-fit: cover;
    width: 100%;
}
```

pfefferle committed:
Refactor site icon image sizing and aspect ratio

Moved site icon image sizing and aspect ratio styles to .site-icon__image class for consistency. Removed redundant styles from actor-switcher and site-hub components, ensuring all icon images use the same sizing and aspect ratio.

pfefferle committed:
Improve site icon layout and positioning

Adjusted the site icon image to use full container size with a dark background for better visual consistency. Updated the dashboard icon button to scale and offset the icon for improved alignment near the viewport edge.

pfefferle commented:
Done.

obenland reviewed (approved):
Thank you! ✨
